wordpress-mobile/WordPress-iOS
# Drop iOS 14 Support: Remove WPButtonForNavigationBar

Related #20860

This is yet another class that triggered a compilation error after bumping the deployment target to iOS 15. It was used in a single place, so I decided to replace it with a regular `UIButton` – just like the rest of the buttons in `PostEditorNavigationBarManager`.

## To test:

- Create a Post
- Check that the Close (X) button is still there

Before:

<img width="875" alt="button-before" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/19634996-bda6-4250-a1a8-3db70d8e69ba">

After:

<img width="793" alt="button-after-ios-16 2" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/416d337b-cacb-47f7-8940-628dba1667f3">


## Regression Notes
1. Potential unintended areas of impact: Create Post
2. What I did to test those areas of impact (or what existing automated tests I relied on): manual testing
3. What automated tests I added (or what prevented me from doing so): n/a

PR submission checklist:
